cmd/software/workspace.go

```diff
@@ -177,5 +184,17 @@ func workspaceSwitch(cmd *cobra.Command, out io.Writer, args []string) error {
 		id = args[0]
 	}
 
-	return workspace.Switch(id, houstonClient, out)
+	pageSize := config.CFG.PageSize.GetInt()
+
+	if config.CFG.Interactive.GetBool() || workspacePaginated {
+		if workspacePageSize < 0 && pageSize > 0 {
+			workspacePageSize = pageSize
+		}
+
+		if !(workspacePageSize > 0 && workspacePageSize <= defaultPageSize) {
+			workspacePageSize = defaultPageSize
```

Comment:
Just slightly confused here, why do we have a `defaultPageSize` in the code and then `pageSize` as a configurable value as well? Shouldn't we fall back to configured pageSize in all cases?

Reply:
I'm not sure how we can restrict at the config level that the page should not be less than 0 and not greater than 100 since houston level we have 100 as the max limit.

its if fine to let user set any number at the CLI level and if invalid houston will throw the error & will be shown in CLI

Reply:
maybe then we can add a warning message if we are using a value not specified by the user, to avoid confusion, wdyt?
Ex: `logrus.Warn("Page size cannot be more than 100, reducing the page size to 100")` or whatever you may think appropriate